src/tsk_fs_file.rs

```diff
@@ -134,6 +136,54 @@ impl<'fs> TskFsFile<'fs> {
         let meta = unsafe { (*self.tsk_fs_file_ptr).meta };
         unsafe{*meta}.type_ & tsk::TSK_FS_META_TYPE_ENUM_TSK_FS_META_TYPE_DIR > 0
     }
+
+    /// Get the TskFsMeta for this TskFsFile
+    pub fn get_meta(&self) -> Result<TskFsMeta, TskError> {
+        TskFsMeta::from_ptr(unsafe{(*self.tsk_fs_file_ptr).meta})
+    }
+
+    /// Get an iterator that reads the file in chunks of buf_size.
+    pub fn read_iter(&'fs self, buf_size: u64) -> TskFsFileIterator<'fs> {
+        let file_size = self.get_meta().unwrap().size();
+        TskFsFileIterator{
+            tsk_fs_file: self,
+            offset:0,
+            file_size: file_size as u64,
+            buf_size
+        }
+    }
+
+    /// Reads the exact buffer size from the file
+    pub fn read_exact(&self,offset:i64, buff: &mut [u8]) -> Result<u64,TskError>{
```

Comment:
Instead of read functionality being methods of TskFsFile, we will want to instead impl Read and Seek trait for TskFsFile or create a new struct like TskFsFileHandle that implements the Read/Seek trait. If Read and Seek trait is implemented you can pass the struct to any parser that parses based off of Read. For example, TskFsAttr implements Read, so you can use a lib like `mft` to do something like:

```rust
use mft::MftParser;

...
let mft_fh = tsk_fs.file_open_meta(0)
	.expect("Could not open $MFT");

let tsk_attr = mft_fh.get_attr()
	.expect("Unable to get default attribute.");

let mut parser = MftParser::from_read_seek(
  tsk_attr, 
  None
).expect("Error getting parser.");

let number_of_entries = parser.get_entry_count();
println!("number_of_entries: {}", number_of_entries);

for i in 0..number_of_entries {
	let entry = parser.get_entry(i as u64);
	...
}
```

This would also allow you to read a tsk file handle via normal means and write into another file like you would normally and not need something like TskFsFileIterator. The other thing we have to consider and the reason why we will most likely have to create a new struct like `TskFsFileHandle` is because we need the option to specify which attribute/data stream we would want to read from. For example, in the case of NTFS, you can have multiple data streams we would want to read from. A good example is $UsnJrnl, where you have Alternative Data Streams like $UsnJrnl:$Max and $UsnJrnl:$J. Thus, there has to be a mechanic in which you can specify what and how you want to read. Another item to account for is the [read flag](http://sleuthkit.org/sleuthkit/docs/api-docs/4.10.1//tsk__fs_8h.html#a9d4fee71e466b85026235e916ec2bea8) . Because of this, it is a much wiser choice to use the [tsk_fs_file_read_type](http://sleuthkit.org/sleuthkit/docs/api-docs/4.10.1//group__fslib.html#ga603ea8f80f1ffbf6be888aeaa508fc81) function when reading from a TskFile as it give you much more control on what and how to read. For that reason, I think it is better to create a new struct `TskFsFileHandle` that stores what and how reads will occur. `TskFsFile` should have methods therefor to get the `TskFsFileHandle` file from the `TskFsFile` and can either be from defaults or specified attributes and flags.

Reply:
Ok, I like this better. I added a new struct called `TskFsFileHandler` which implements `Read` and `Seek`. The struct accepts `TskFsFile`, `TskFsAttrType` and `TskFsReadFlag`. `TskFsReadFlag ` is an enum and it is specific for the reading so it could be add to the file `tsk_fs_file_handler.rs`. However, the TskFsAttrType enum is the same types you already implemented on the TskFsAttr::type_name(). I think a better way of implementing it is:
* Add `TskFsAttrType` to `tsk_fs_file.rs`.
* Implement `Display` trait for `TskFsAttrType`
* change `type_name()` to use Display implementation.

Let me know if you are ok with these changes.